core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/codec/CompressorPool.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.file.rfile.bcfile.codec;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.file.rfile.bcfile.Compression.Algorithm;
+import org.apache.commons.pool2.impl.GenericKeyedObjectPool;
+import org.apache.hadoop.io.compress.Compressor;
+import org.apache.hadoop.io.compress.Decompressor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Compressor factory extension that enables object pooling using Commons Pool. The design will have
+ * a keyed compressor pool and decompressor pool. The key of which will be the Algorithm itself.
+ *
+ */
+public class CompressorPool extends DefaultCompressorFactory {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CompressorPoolFactory.class);
+
+  /**
+   * Compressor pool.
+   */
+  GenericKeyedObjectPool<Algorithm,Compressor> compressorPool;
+
+  /**
+   * Decompressor pool
+   */
+  GenericKeyedObjectPool<Algorithm,Decompressor> decompressorPool;
+
+  public CompressorPool(AccumuloConfiguration acuConf) {
+
+    super(acuConf);
+
+    compressorPool = new GenericKeyedObjectPool<Algorithm,Compressor>(new CompressorPoolFactory());
```

Comment:
I looked at the commons pool javadoc a bit and noticed it had a GenericKeyedObjectPoolConfig object.  Not sure if this is workable, but I was wondering about making the config pass through.  Where a prop could be set in the Accumulo config for any setter on GenericKeyedObjectPoolConfig.  Like GenericKeyedObjectPoolConfig has a setMaxTotal method. The accumulo config could have something like.

```
tserver.compressor.factory.class=...CompressorPool
tserver.compressor.factory.opts.maxTotal=42
```

and here when it sees maxTotal it calls the setMaxTotal method on GenericKeyedObjectPoolConfig.